2022-12 Release (#95)
ni/install-systemlink-enterprise: release-notes/2022-12/README.md

# SystemLink Enterprise 2022-12 Release Notes

The 2022-12 release bundle for SystemLink Enterprise has been published to <https://niedge01.jfrog.io>. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

## New Features and Behavior changes

> - The License service now included in the top level helm chart. Refer to **Helm Chart Breaking Changes** for details on the required configuration for this service.

Reviewer: ```suggestion
- The top level helm chart includes the License service. Refer to **Helm Chart Breaking Changes** for details on the required configuration for this service.
```

- Users can specify data table IDs as variables in dashboards.

> - Users can customize columns, filter, save, and load views in the Products grid.

Reviewer: ```suggestion
- Users can customize, filter, save, and load views in the Products grid.
```
I know it's not as specific but it's easier to read and, I think, still gets the point across

        - Refer to [systemlink-values.yaml](https://github.com/ni/install-systemlink-enterprise/blob/main/getting-started/templates/systemlink-values.yaml)
        - Refer to [What Do You Need to Use SystemLink Enterprise?](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/systemlink-enterprise-requirements.html)
- `nbexecservice 0.1.312`
    - Config map renamed from `nbexec-service-argo-configmap` to `<helm_release_name>-nbexecservice-argo-configmap`.
    - Service names were moved from config map to workflow-template.
    - During helm upgrade in progress executions may fail.
- `routineservice 0.2.0`
    - The contents of the `routineservice` helm chart have been split into two charts": `routineservice` and `routineeventtrigger`
    - There are no breaking changes to the existing chart but some values and secrets can be removed or renamed.
        - Within `userservices.secrets.whitelistedApiKeys`, `routines` can be removed and `routineeventtrigger` should be added.
        - The `routineservice.engine` configuration is deprecated. If you used this configuration, migrate by moving it under `routineeventtrigger`.
- `routineeventtrigger 0.2.0`

>     - Some components from the "routineservice" into the `routineeventtrigger` chart.

Reviewer 2: I don't remember what I was trying to write here but this needs to be reformulated

Author: I commented out this section for now. It appears to be redundant with the bullets on routine related services below. Regardless I will defer to you on the accuracy of this information.

Author: Changed

    - Refer to [systemlink-secrets.yaml](TODO Link) for required changes
- `routinescheduletrigger 0.2.0`
    - Newly included chart
    - You must configure the `userservices.secrets.whitelistedApiKeys` value with an entry for `routinescheduletrigger` `serviceName`.
    - You must configure MongoDB credentials in `routinescheduletrigger.secrets.mongodb`.
- `routineexecutor 0.2.0`
    - Newly included chart

>     - You must configure the `userservices.secrets.whitelistedApiKeys` value with a n entry for `routineexecutor` `serviceName`.

Reviewer 2: ```suggestion
    - You must configure the `userservices.secrets.whitelistedApiKeys` value with an entry for `routineexecutor` `serviceName`.
```
